src/commands/cmd_bloom_filter.cc

```diff
@@ -95,11 +95,46 @@ class CommandBFAdd : public Commander {
     auto s = bloom_db.Add(args_[1], args_[2], &ret);
     if (!s.ok()) return {Status::RedisExecErr, s.ToString()};
 
-    *output = redis::Integer(ret);
+    if (ret == -1) {
+      *output = redis::Error("ERR nonscaling filter is full");
+    } else {
+      *output = redis::Integer(ret);
+    }
     return Status::OK();
   }
 };
 
+class CommandBFMAdd : public Commander {
+ public:
+  Status Parse(const std::vector<std::string> &args) override {
+    items_.reserve(args_.size() - 2);
+    for (size_t i = 2; i < args_.size(); ++i) {
+      items_.emplace_back(args_[i]);
+    }
+    return Commander::Parse(args);
+  }
+
+  Status Execute(Server *svr, Connection *conn, std::string *output) override {
+    redis::BloomChain bloom_db(svr->storage, conn->GetNamespace());
+    std::vector<int> rets(items_.size(), 0);
```

Comment:
Would something like `vector<bool>` better here?

Reply:
Sure, I think it's necessary to change it(and `Exists`) to the explicit type and name.